### PR TITLE
feature: conn param in experiments() function

### PR DESCRIPTION
**Pull request by thibaudruelle**
Makes it possible to list the experiments contained in any database, not just the one in the configuration file.

Changes proposed in this pull request:
- add an optional `conn` param to the `experiments()` and `load_experiment()` functions in experiment_container.py

@GateBuilder @astafan8

### CODE REVIEW

File: qcodes/dataset/experiment_container.py [hunk 1]
```diff
@@ -180,18 +180,23 @@ def __repr__(self) -> str:
...
+    Args:
+        conn: connection to the database. If not supplied, a new connection
+          to the DB file specified in the config is made
+
     Returns:
         All the experiments in the container
     """
-    log.info("loading experiments from {}".format(get_DB_location()))
-    rows = get_experiments(connect(get_DB_location(), get_DB_debug()))
+    log.info("loading experiments from {}".format(conn))
```
Comment (astafan8) [inline qcodes/dataset/experiment_container.py]: to my knowledge printing the connection object is quite useless. However, `ConnectionPlus` object has `.path_to_dbfile` attribute which can be used for this log message after the next line is executed. (i mean to suggest to move this logging line after the next line)

**Comment by astafan8** [comment]
@thibaudruelle I'm happy to make the requested changes myself, but I don't have the rights to "edit the files in this PR here, on GitHub webpage" (and likely I don't have rights to push to your branch) - likely because I'm not a "collaborator" on your fork. So you could add me as a collaborator or I can open a new duplicate PR?

**Comment by thibaudruelle** [comment]
@astafan8 Sorry I overlooked this. I made the changes.

**Comment by astafan8** [comment]
@thibaudruelle could you update the branch so that these changes are on top of qcodes master? Otherwise github is saying "This branch is out-of-date with the base branch"

**Comment by thibaudruelle** [comment]
@astafan8 I merged from my local repo, should probably have hit the big update button instead ... I hope it is okay like that.

**Comment by jenshnielsen** [comment]
@thibaudruelle This looks fine. Its ready to merge once travis and pipelines passes

**Pull request merged**